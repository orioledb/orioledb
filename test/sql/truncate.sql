CREATE SCHEMA truncate;
SET SESSION search_path = 'truncate';
CREATE EXTENSION orioledb;

CREATE VIEW o_relnames AS
    SELECT c.relname FROM (
        SELECT reloid, relnode FROM orioledb_table_oids()
        UNION
        SELECT index_reloid, index_relnode FROM orioledb_index_oids()) ot
        JOIN pg_class c ON c.oid = ot.reloid
        WHERE c.relfilenode = ot.relnode
        ORDER BY relname;

CREATE TABLE o_non_transactional_truncate (
  a int UNIQUE,
  b char
) USING orioledb;

BEGIN;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate%';
INSERT INTO o_non_transactional_truncate VALUES (3, 'b');
TRUNCATE o_non_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate%';
COMMIT;

BEGIN;

CREATE TABLE o_transactional_truncate (
  a int UNIQUE,
  b char
) USING orioledb;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate%';
INSERT INTO o_transactional_truncate VALUES (3, 'b');
TRUNCATE o_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate%';
COMMIT;

CREATE TABLE o_non_transactional_truncate_pkey (
  a int PRIMARY KEY,
  b char UNIQUE
) USING orioledb;

BEGIN;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate_pkey%';
INSERT INTO o_non_transactional_truncate VALUES (3, 'b');
TRUNCATE o_non_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate_pkey%';
COMMIT;

BEGIN;

CREATE TABLE o_transactional_truncate_pkey (
  a int PRIMARY KEY,
  b char UNIQUE
) USING orioledb;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate_pkey%';
INSERT INTO o_transactional_truncate_pkey VALUES (3, 'b');
TRUNCATE o_transactional_truncate_pkey;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate_pkey%';
COMMIT;

DROP EXTENSION orioledb CASCADE;
DROP SCHEMA truncate CASCADE;
RESET search_path;
