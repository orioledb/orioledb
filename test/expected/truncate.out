CREATE SCHEMA truncate;
SET SESSION search_path = 'truncate';
CREATE EXTENSION orioledb;
CREATE VIEW o_relnames AS
    SELECT c.relname FROM (
        SELECT reloid, relnode FROM orioledb_table_oids()
        UNION
        SELECT index_reloid, index_relnode FROM orioledb_index_oids()) ot
        JOIN pg_class c ON c.oid = ot.reloid
        WHERE c.relfilenode = ot.relnode
        ORDER BY relname;
CREATE TABLE o_non_transactional_truncate (
  a int UNIQUE,
  b char
) USING orioledb;
BEGIN;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate%';
              relname               
------------------------------------
 o_non_transactional_truncate
 o_non_transactional_truncate_a_key
(2 rows)

INSERT INTO o_non_transactional_truncate VALUES (3, 'b');
TRUNCATE o_non_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate%';
              relname               
------------------------------------
 o_non_transactional_truncate
 o_non_transactional_truncate_a_key
(2 rows)

COMMIT;
BEGIN;
CREATE TABLE o_transactional_truncate (
  a int UNIQUE,
  b char
) USING orioledb;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate%';
            relname             
--------------------------------
 o_transactional_truncate
 o_transactional_truncate_a_key
(2 rows)

INSERT INTO o_transactional_truncate VALUES (3, 'b');
TRUNCATE o_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate%';
            relname             
--------------------------------
 o_transactional_truncate
 o_transactional_truncate_a_key
(2 rows)

COMMIT;
CREATE TABLE o_non_transactional_truncate_pkey (
  a int PRIMARY KEY,
  b char UNIQUE
) USING orioledb;
BEGIN;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate_pkey%';
                 relname                 
-----------------------------------------
 o_non_transactional_truncate_pkey
 o_non_transactional_truncate_pkey_b_key
 o_non_transactional_truncate_pkey_pkey
(3 rows)

INSERT INTO o_non_transactional_truncate VALUES (3, 'b');
TRUNCATE o_non_transactional_truncate;
SELECT * FROM o_relnames WHERE relname LIKE 'o_non_transactional_truncate_pkey%';
                 relname                 
-----------------------------------------
 o_non_transactional_truncate_pkey
 o_non_transactional_truncate_pkey_b_key
 o_non_transactional_truncate_pkey_pkey
(3 rows)

COMMIT;
BEGIN;
CREATE TABLE o_transactional_truncate_pkey (
  a int PRIMARY KEY,
  b char UNIQUE
) USING orioledb;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate_pkey%';
               relname               
-------------------------------------
 o_transactional_truncate_pkey
 o_transactional_truncate_pkey_b_key
 o_transactional_truncate_pkey_pkey
(3 rows)

INSERT INTO o_transactional_truncate_pkey VALUES (3, 'b');
TRUNCATE o_transactional_truncate_pkey;
SELECT * FROM o_relnames WHERE relname LIKE 'o_transactional_truncate_pkey%';
               relname               
-------------------------------------
 o_transactional_truncate_pkey
 o_transactional_truncate_pkey_b_key
 o_transactional_truncate_pkey_pkey
(3 rows)

COMMIT;
DROP EXTENSION orioledb CASCADE;
NOTICE:  drop cascades to 5 other objects
DETAIL:  drop cascades to table o_non_transactional_truncate
drop cascades to table o_transactional_truncate
drop cascades to table o_non_transactional_truncate_pkey
drop cascades to table o_transactional_truncate_pkey
drop cascades to view o_relnames
DROP SCHEMA truncate CASCADE;
RESET search_path;
